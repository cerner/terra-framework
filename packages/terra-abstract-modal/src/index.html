<!doctype html>
<html dir="ltr" lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="x-ua-compatible" content="ie=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>Abstract Modal Test</title>
  </head>
  <body>
    <div id="root"></div>
  </body>
</html>