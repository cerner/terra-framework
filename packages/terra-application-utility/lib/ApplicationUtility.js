"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ApplicationHeaderUtility", {
  enumerable: true,
  get: function get() {
    return _ApplicationHeaderUtility.default;
  }
});
Object.defineProperty(exports, "ApplicationMenuUtility", {
  enumerable: true,
  get: function get() {
    return _ApplicationMenuUtility.default;
  }
});
Object.defineProperty(exports, "UtilityUtils", {
  enumerable: true,
  get: function get() {
    return _Utils.default;
  }
});
exports.default = void 0;
var _ApplicationHeaderUtility = _interopRequireDefault(require("./ApplicationHeaderUtility"));
var _ApplicationMenuUtility = _interopRequireDefault(require("./ApplicationMenuUtility"));
var _Utils = _interopRequireDefault(require("./Utils"));
var ApplicationUtility = {
  ApplicationHeaderUtility: _ApplicationHeaderUtility.default,
  ApplicationMenuUtility: _ApplicationMenuUtility.default,
  UtilityUtils: _Utils.default
};
var _default = ApplicationUtility;
exports.default = _default;