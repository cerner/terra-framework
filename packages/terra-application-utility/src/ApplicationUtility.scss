:local {
  .application-utility {
    /*TODO: Add styles */
  }
}
