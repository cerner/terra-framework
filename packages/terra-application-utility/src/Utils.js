import PropTypes from 'prop-types';

const KEY_CODES = {
  ENTER: 13,
  SPACE: 32,
  UP_ARROW: 38,
  DOWN_ARROW: 40,
  LEFT_ARROW: 37,
  RIGHT_ARROW: 39,
  BACK_SPACE: 8,
  TAB: 9,
};

const LOCATIONS = {
  BODY: 'body',
  FOOTER: 'footer',
};

const VARIANTS = {
  HEADER: 'header',
  MENU: 'menu',
};

const itemShape = PropTypes.shape({
  /**
   * Array containing the keys of each child item of this item.
   */
  childKeys: PropTypes.arrayOf(PropTypes.string),
  /**
   * The component associated with this item.
   */
  content: PropTypes.object,
  /**
   * The location to place the item. One of Utils.LOCATIONS.BODY, Utils.LOCATIONS.FOOTER.
   */
  contentLocation: PropTypes.oneOf([LOCATIONS.BODY, LOCATIONS.FOOTER]),
  /**
   * Boolean indicating if the item is read-only. Takes precedence over isSelected/Selectable.
   */
  isReadOnly: PropTypes.bool,
  /**
   * Boolean indicating if the item is selected.
   */
  isSelected: PropTypes.bool,
  /**
   * Boolean indicating if the item is selectable.
   */
  isSelectable: PropTypes.bool,
  /**
   * The unique key associated with this item.
   */
  key: PropTypes.string.isRequired,
  /**
   * Optional meta data to be returned along with the item key within the onChange.
   */
  metaData: PropTypes.object,
  /**
   * The text associated with this item.
   */
  title: PropTypes.string,
});

const Utils = {
  KEY_CODES,
  LOCATIONS,
  VARIANTS,
  itemShape,
};

export default Utils;
