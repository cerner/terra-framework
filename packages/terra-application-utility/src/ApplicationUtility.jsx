import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import styles from './ApplicationUtility.scss';

const cx = classNames.bind(styles);

const propTypes = {
 /*
 * Content to be displayed as the name
 */
  name: PropTypes.string,
};

const defaultProps = {
  name: 'default',
};

const ApplicationUtility = ({ name, ...customProps }) => {
  const attributes = Object.assign({}, customProps);
  const ApplicationUtilityClassNames = cx([
    'application-utility',
    attributes.className,
  ]);

  return (<div {...attributes} className={ApplicationUtilityClassNames} >{name}</div>);
};

ApplicationUtility.propTypes = propTypes;
ApplicationUtility.defaultProps = defaultProps;

export default ApplicationUtility;
