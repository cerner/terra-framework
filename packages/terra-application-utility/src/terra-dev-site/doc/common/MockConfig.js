import { UtilityUtils } from '../../../ApplicationUtility';

const MockConfig = customComponent => [
  {
    key: 'menu',
    title: 'Menu',
    contentLocation: UtilityUtils.LOCATIONS.BODY,
    isSelected: false,
    isSelectable: false,
    metaData: { otherInformation: 'Other Information' },
    childKeys: [
      'item-1',
      'item-2',
      'item-3',
      'item-4',
      'item-5',
      'item-6',
      'item-7',
      'item-8',
    ],
  },
  {
    key: 'item-1',
    title: 'Item 1',
    content: customComponent,
    contentLocation: UtilityUtils.LOCATIONS.BODY,
    isSelected: false,
    isSelectable: false,
    childKeys: [
      'item-1.1',
    ],
  },
  {
    key: 'item-2',
    title: 'Item 2',
    isSelected: false,
    isSelectable: false,
    childKeys: [
      'item-2.1',
      'item-2.2',
    ],
    id: 'test-item-2',
  },
  {
    key: 'item-3',
    title: 'item-3',
    contentLocation: UtilityUtils.LOCATIONS.FOOTER,
    isSelected: false,
    isSelectable: false,
    childKeys: [],
  },
  {
    key: 'item-4',
    title: 'Item 4',
    contentLocation: UtilityUtils.LOCATIONS.BODY,
    isSelected: false,
    isSelectable: false,
    childKeys: [
      'item-4.1',
      'item-4.2',
      'item-4.3',
    ],
  },
  {
    key: 'item-1.1',
    title: 'Item 1.1',
    contentLocation: UtilityUtils.LOCATIONS.BODY,
    isSelected: false,
    isSelectable: false,
    metaData: { otherInformation: 'Other Information' },
    childKeys: [],
  },
  {
    key: 'item-2.1',
    title: 'Item 2.1',
    contentLocation: UtilityUtils.LOCATIONS.BODY,
    isSelected: false,
    isSelectable: false,
    childKeys: [],
  },
  {
    key: 'item-2.2',
    title: 'Item 2.2',
    contentLocation: UtilityUtils.LOCATIONS.BODY,
    isSelected: false,
    isSelectable: false,
    childKeys: [],
  },
  {
    key: 'item-4.1',
    title: 'Item 4.1',
    contentLocation: UtilityUtils.LOCATIONS.BODY,
    isSelected: false,
    isSelectable: false,
    childKeys: [],
  },
  {
    key: 'item-4.2',
    title: 'Item 4.2',
    contentLocation: UtilityUtils.LOCATIONS.BODY,
    isSelected: false,
    isSelectable: false,
    childKeys: [],
  },
  {
    key: 'item-4.3',
    title: 'Item 4.3',
    contentLocation: UtilityUtils.LOCATIONS.BODY,
    isSelected: false,
    isSelectable: false,
    childKeys: [],
  },
  {
    key: 'item-5',
    title: 'Item 5',
    contentLocation: UtilityUtils.LOCATIONS.BODY,
    isSelected: false,
    isSelectable: false,
    childKeys: ['item-5.1', 'item-5.2'],
  },
  {
    key: 'item-5.1',
    title: 'Item 5.1',
    contentLocation: UtilityUtils.LOCATIONS.BODY,
    isSelected: false,
    isSelectable: true,
    childKeys: [],
  },
  {
    key: 'item-5.2',
    title: 'Item 5.2',
    contentLocation: UtilityUtils.LOCATIONS.BODY,
    isSelected: false,
    isSelectable: true,
    childKeys: [],
  },
  {
    key: 'item-6',
    title: 'Item 6',
    contentLocation: UtilityUtils.LOCATIONS.FOOTER,
    isSelected: false,
    isSelectable: true,
    childKeys: [],
  },
  {
    key: 'item-7',
    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua',
    contentLocation: UtilityUtils.LOCATIONS.BODY,
    isSelected: true,
    isSelectable: false,
    childKeys: [],
    id: 'test-item-7',
  },
  {
    key: 'item-8',
    title: 'Read Only Item',
    contentLocation: UtilityUtils.LOCATIONS.BODY,
    isReadOnly: true,
    isSelected: false,
    isSelectable: false,
    childKeys: [],
    id: 'test-item-8',
  },
];

export default MockConfig;
