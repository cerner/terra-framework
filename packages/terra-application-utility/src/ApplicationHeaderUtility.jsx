import React from 'react';
import PropTypes from 'prop-types';
import 'terra-base/lib/baseStyles';
import Utils from './Utils';
import UtilityButton from './utility/_UtilityButton';

const propTypes = {
  /**
   * The array containing the menu items to be rendered within the menu.
   */
  menuItems: PropTypes.arrayOf(Utils.itemShape).isRequired,
  /**
   * The function to trigger when a menu item is selected.
   * Returns `(event, { key: String, metaData: Object})`
   */
  onChange: PropTypes.func.isRequired,
  /**
   * The function that discloses the menu.
   */
  onDisclose: PropTypes.func.isRequired,
  /**
   * The key of the top level menu page.
   */
  initialSelectedKey: PropTypes.string.isRequired,
  /**
   * The text to be displayed.
   */
  title: PropTypes.string,
  /**
   * The accessory element to be displayed next to the title.
   */
  accessory: PropTypes.element,
};

const ApplicationHeaderUtility = ({
  menuItems,
  onChange,
  onDisclose,
  initialSelectedKey,
  title,
  accessory,
  ...customProps
}) => (
  <UtilityButton
    {...customProps}
    menuItems={menuItems}
    onChange={onChange}
    onDisclose={onDisclose}
    initialSelectedKey={initialSelectedKey}
    title={title}
    accessory={accessory}
    variant={Utils.VARIANTS.HEADER}
  />
);

ApplicationHeaderUtility.propTypes = propTypes;

export default ApplicationHeaderUtility;
