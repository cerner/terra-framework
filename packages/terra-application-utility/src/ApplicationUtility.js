import ApplicationHeaderUtility from './ApplicationHeaderUtility';
import ApplicationMenuUtility from './ApplicationMenuUtility';
import UtilityUtils from './Utils';

const ApplicationUtility = {
  ApplicationHeaderUtility,
  ApplicationMenuUtility,
  UtilityUtils,
};

export default ApplicationUtility;
export { ApplicationHeaderUtility, ApplicationMenuUtility, UtilityUtils };
