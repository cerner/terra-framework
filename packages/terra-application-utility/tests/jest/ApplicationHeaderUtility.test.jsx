import React from 'react';
import { ApplicationHeaderUtility } from '../../src/ApplicationUtility';
import MockConfig from '../../src/terra-dev-site/doc/common/MockConfig';

describe('ApplicationHeaderUtility', () => {
  const mockOnChange = jest.fn();
  const mockOnDisclose = jest.fn();
  const mockOnRequestClose = jest.fn();
  const initialSelectedKey = 'menu';
  const accessory = <image />;
  const title = 'name';

  it('should render with default props', () => {
    const wrapper = shallow(<ApplicationHeaderUtility
      menuItems={[]}
      onChange={mockOnChange}
      onDisclose={mockOnDisclose}
      onRequestClose={mockOnRequestClose}
      initialSelectedKey={initialSelectedKey}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a title', () => {
    const wrapper = shallow(<ApplicationHeaderUtility
      menuItems={[]}
      onChange={mockOnChange}
      onDisclose={mockOnDisclose}
      onRequestClose={mockOnRequestClose}
      initialSelectedKey={initialSelectedKey}
      title={title}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with an accessory', () => {
    const wrapper = shallow(<ApplicationHeaderUtility
      menuItems={[]}
      onChange={mockOnChange}
      onDisclose={mockOnDisclose}
      onRequestClose={mockOnRequestClose}
      initialSelectedKey={initialSelectedKey}
      accessory={accessory}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with menu items', () => {
    const wrapper = shallow(<ApplicationHeaderUtility
      menuItems={MockConfig()}
      onChange={mockOnChange}
      onDisclose={mockOnDisclose}
      onRequestClose={mockOnRequestClose}
      initialSelectedKey={initialSelectedKey}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should pass in a custom prop', () => {
    const wrapper = shallow(<ApplicationHeaderUtility
      menuItems={[]}
      onChange={mockOnChange}
      onDisclose={mockOnDisclose}
      onRequestClose={mockOnRequestClose}
      initialSelectedKey={initialSelectedKey}
      id="test"
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a user specified role', () => {
    const menuRole = 'menu';
    const wrapper = shallow(<ApplicationHeaderUtility
      menuItems={[]}
      onChange={mockOnChange}
      onDisclose={mockOnDisclose}
      onRequestClose={mockOnRequestClose}
      initialSelectedKey={initialSelectedKey}
      title={title}
      menuRole={menuRole}
    />);
    expect(wrapper.prop('menuRole')).toBe(menuRole);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a default role when no role is specified', () => {
    const defaultMenuRole = 'navigation';
    const wrapper = shallow(<ApplicationHeaderUtility
      menuItems={[]}
      onChange={mockOnChange}
      onDisclose={mockOnDisclose}
      onRequestClose={mockOnRequestClose}
      initialSelectedKey={initialSelectedKey}
      title={title}
      menuRole={defaultMenuRole}
    />);
    expect(wrapper.prop('menuRole')).toBe(defaultMenuRole);
    expect(wrapper).toMatchSnapshot();
  });
});
