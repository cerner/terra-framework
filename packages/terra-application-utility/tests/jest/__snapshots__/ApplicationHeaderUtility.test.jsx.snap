// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationHeaderUtility it should pass in a custom prop 1`] = `
<InjectIntl(UtilityButton)
  id="test"
  initialSelectedKey="menu"
  menuItems={[]}
  menuRole="navigation"
  onChange={[MockFunction]}
  onDisclose={[MockFunction]}
  onRequestClose={[MockFunction]}
  variant="header"
/>
`;

exports[`ApplicationHeaderUtility should render with a default role when no role is specified 1`] = `
<InjectIntl(UtilityButton)
  initialSelectedKey="menu"
  menuItems={[]}
  menuRole="navigation"
  onChange={[MockFunction]}
  onDisclose={[MockFunction]}
  onRequestClose={[MockFunction]}
  title="name"
  variant="header"
/>
`;

exports[`ApplicationHeaderUtility should render with a title 1`] = `
<InjectIntl(UtilityButton)
  initialSelectedKey="menu"
  menuItems={[]}
  menuRole="navigation"
  onChange={[MockFunction]}
  onDisclose={[MockFunction]}
  onRequestClose={[MockFunction]}
  title="name"
  variant="header"
/>
`;

exports[`ApplicationHeaderUtility should render with a user specified role 1`] = `
<InjectIntl(UtilityButton)
  initialSelectedKey="menu"
  menuItems={[]}
  menuRole="menu"
  onChange={[MockFunction]}
  onDisclose={[MockFunction]}
  onRequestClose={[MockFunction]}
  title="name"
  variant="header"
/>
`;

exports[`ApplicationHeaderUtility should render with an accessory 1`] = `
<InjectIntl(UtilityButton)
  accessory={<image />}
  initialSelectedKey="menu"
  menuItems={[]}
  menuRole="navigation"
  onChange={[MockFunction]}
  onDisclose={[MockFunction]}
  onRequestClose={[MockFunction]}
  variant="header"
/>
`;

exports[`ApplicationHeaderUtility should render with default props 1`] = `
<InjectIntl(UtilityButton)
  initialSelectedKey="menu"
  menuItems={[]}
  menuRole="navigation"
  onChange={[MockFunction]}
  onDisclose={[MockFunction]}
  onRequestClose={[MockFunction]}
  variant="header"
/>
`;

exports[`ApplicationHeaderUtility should render with menu items 1`] = `
<InjectIntl(UtilityButton)
  initialSelectedKey="menu"
  menuItems={
    [
      {
        "childKeys": [
          "item-1",
          "item-2",
          "item-3",
          "item-4",
          "item-5",
          "item-6",
          "item-7",
          "item-8",
        ],
        "contentLocation": "body",
        "isSelectable": false,
        "isSelected": false,
        "key": "menu",
        "metaData": {
          "otherInformation": "Other Information",
        },
        "title": "Menu",
      },
      {
        "childKeys": [
          "item-1.1",
        ],
        "content": undefined,
        "contentLocation": "body",
        "isSelectable": false,
        "isSelected": false,
        "key": "item-1",
        "title": "Name",
      },
      {
        "childKeys": [
          "item-2.1",
          "item-2.2",
        ],
        "id": "test-item-2",
        "isSelectable": false,
        "isSelected": false,
        "key": "item-2",
        "title": "Profile",
      },
      {
        "childKeys": [],
        "contentLocation": "footer",
        "isSelectable": false,
        "isSelected": false,
        "key": "item-3",
        "title": "Notifications",
      },
      {
        "childKeys": [
          "item-4.1",
          "item-4.2",
          "item-4.3",
        ],
        "contentLocation": "body",
        "isSelectable": false,
        "isSelected": false,
        "key": "item-4",
        "title": "Projects",
      },
      {
        "childKeys": [],
        "contentLocation": "body",
        "isSelectable": false,
        "isSelected": false,
        "key": "item-1.1",
        "metaData": {
          "otherInformation": "Other Information",
        },
        "title": "John Doe",
      },
      {
        "childKeys": [],
        "contentLocation": "body",
        "isSelectable": false,
        "isSelected": false,
        "key": "item-2.1",
        "title": "Gender: Male",
      },
      {
        "childKeys": [],
        "contentLocation": "body",
        "isSelectable": false,
        "isSelected": false,
        "key": "item-2.2",
        "title": "Location: Germany",
      },
      {
        "childKeys": [],
        "contentLocation": "body",
        "isSelectable": false,
        "isSelected": false,
        "key": "item-4.1",
        "title": "Project 1",
      },
      {
        "childKeys": [],
        "contentLocation": "body",
        "isSelectable": false,
        "isSelected": false,
        "key": "item-4.2",
        "title": "Project 2",
      },
      {
        "childKeys": [],
        "contentLocation": "body",
        "isSelectable": false,
        "isSelected": false,
        "key": "item-4.3",
        "title": "Project 3",
      },
      {
        "childKeys": [
          "item-5.1",
          "item-5.2",
        ],
        "contentLocation": "body",
        "isSelectable": false,
        "isSelected": false,
        "key": "item-5",
        "title": "Skills",
      },
      {
        "childKeys": [],
        "contentLocation": "body",
        "isSelectable": true,
        "isSelected": false,
        "key": "item-5.1",
        "title": "Skill 1",
      },
      {
        "childKeys": [],
        "contentLocation": "body",
        "isSelectable": true,
        "isSelected": false,
        "key": "item-5.2",
        "title": "Skill 2",
      },
      {
        "childKeys": [],
        "contentLocation": "footer",
        "isSelectable": true,
        "isSelected": false,
        "key": "item-6",
        "title": "Logout",
      },
      {
        "childKeys": [],
        "contentLocation": "body",
        "id": "test-item-7",
        "isSelectable": false,
        "isSelected": true,
        "key": "item-7",
        "title": "Drill-down capabilities provide the ability for the end-user to access different levels of information without creating ad hoc queries for each level of information. Instead the user is able to drill down to a detail level, or elevate to the summary level, through pull-down menus or mouse clicks.",
      },
      {
        "childKeys": [],
        "contentLocation": "body",
        "id": "test-item-8",
        "isReadOnly": true,
        "isSelectable": false,
        "isSelected": false,
        "key": "item-8",
        "title": "Last logged in: yesterday",
      },
    ]
  }
  menuRole="navigation"
  onChange={[MockFunction]}
  onDisclose={[MockFunction]}
  onRequestClose={[MockFunction]}
  variant="header"
/>
`;
